### PR TITLE
feat: Add support for llamaIndex in evaluation

### PR DESCRIPTION
**Pull request by suekou**
Added type checks for llamaIndex LLMs and embeddings in the evaluate function.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -8,6 +8,7 @@ dependencies = [
     "langchain-core",
     "langchain-community",
     "langchain_openai",
+    "llama_index",
```
Comment (jjmachan) [inline pyproject.toml:11]: llamaindex is actually not part of the core dependency. It's best if we keep it optional

**Review by jjmachan — COMMENTED** [review]
thanks a lot for sending a PR 🙂 - just once change though and we'll merge it in

File: pyproject.toml (hunk 1 above)
Comment (jjmachan) [reply pyproject.toml:11]: if you move it to `all` that would be helpful, or maybe even a new group for llamaindex?

Comment (suekou) [reply pyproject.toml]: Thanks for the feedback! I've updated the PR to make llama_index optional by moving it to the [all] group.

**Review by jjmachan — APPROVED** [review]
Thanks a lot @suekou 🙂 ❤️ 
merging this in

**Comment by jjmachan** [comment]
btw would love to get your feedback on this docs too if you get the chance https://docs.ragas.io/en/latest/howtos/integrations/_llamaindex/

**Comment by suekou** [comment]
@jjmachan I found this to be very clear and understandable! 
I noticed the URL on the page was outdated, so I submitted a PR to update it!

**Comment by jjmachan** [comment]
hey @suekou thats a good point but right now `evaluate` in ragas.integrations.llamaindex has the same features as the top level evaluate right?

the idea was to keep both of the feature compatible since behind the scene all we are doing is running the query_engine for you

**Pull request merged**